lib/catalog/rbac/access.rb

```diff
@@ -14,55 +14,48 @@ def read_access_check
         resource_check('read')
       end
 
-      def create_access_check
-        permission_check('create')
+      def create_access_check(klass)
+        permission_check('create', klass)
       end
 
       def destroy_access_check
         resource_check('delete')
       end
 
-      def admin_check
-        return true unless rbac_enabled?
-
-        catalog_admin?
-      end
-
       def resource_check(verb, id = @record.id, klass = @record.class)
         return true unless rbac_enabled?
-        return true if catalog_admin?
 
-        return false unless access_object(@record.class.table_name, verb).accessible?
-        ids = access_id_list(verb, klass)
-        return false if klass.try(:supports_access_control?) && ids.exclude?(id.to_s)
+        scopes = access_object.scopes(@record.class.table_name, verb)
+        check = if scopes.include?("admin")
```

Comment:
@eclarizio We could remove the check variable and the return value will be the value from the if block. Since we have a false at the end. Technically nothing should hit the else clause. Can we just add an error message to the else clause it will be easier to debug if we accidentally land there for some object.

Reply:
Sure, that's fine :+1: